File: C#Basics/07. CSharpAdvancedTopics/2. PrimeChecker/PrimeChecker.cs
//Write a Boolean method IsPrime(n) that check whether a given integer number n is prime.

using System;

class PrimeChecker
{
    static void Main()
    {
        Console.WriteLine("Please, type an integer number N and press Enter:");
        int n = int.Parse(Console.ReadLine());

        Console.WriteLine(PrimeChecker.IsPrime(n));
    }

    private static bool IsPrime(int n)
    {
        int divider = 1;
        int count = 0;

        while (divider < Math.Sqrt(n))
        {
            if (n % divider == 0)
            {
                count++;
            }
            divider++;
        }

        if (count < 2)
        {
            return true;
        }
        else
        {
            return false;
        }
    }
}

File: JavascriptBasics/02. JavaScriptSyntax/04. CheckIfNumberIsPrime/primeChecker.js
//Write a JavaScript function isPrime(value) that checks if an integer number is prime.
//Write JS program primeChecker.js that checks if a few numbers are prime.
//The result should be printed on the console (true or false) on the console.
//Run the program through Node.js.

'use strict';

function isPrime(value) {
	var integerNumber = value || 0,
		loopLimit = Math.sqrt(integerNumber),
		isPrimeNumber = true,
		divisionResult;

	if (integerNumber > 2) {
		for (var i = 2; i < loopLimit; i++) {
			if (integerNumber % i === 0) {
				isPrimeNumber = false;
				break;
			}
		}
	}

	console.log(isPrimeNumber);
}

isPrime(7);
isPrime(254);
isPrime(587);